src/ip.py

```diff
@@ -185,45 +190,69 @@ def do_route_list(af):
         dev = ra[5]
       if flags.find('W') != -1:
         continue
+      if flags.find('B') >= 0:
+        blackholes.append(ra)
+        continue
       if target == 'default':
         print('default via ' + gw + ' dev ' + dev)
       else:
-        dots=target.count('.')
-        if target.find('/') == -1:
-          addr=target
-          netmask=8+dots*8
-        else:
-          [addr,netmask] = target.split('/')
-
-        if dots == 2:
-          addr = addr + '.0'
-        elif dots == 1:
-          addr = addr + '.0.0'
-        elif dots == 0:
-          addr = addr + '.0.0.0'
-
+        cidr = cidr_from_netstat_target(target)
         if re.match("link.+",gw):
-          print(addr + '/' + str(netmask)+ ' dev ' + dev + '  scope link')
+          print(cidr + ' dev ' + dev + '  scope link')
         else:
-          print(addr + '/' + str(netmask) + ' via ' + gw + ' dev ' + dev)
+          print(cidr + ' via ' + gw + ' dev ' + dev)
+  for b in blackholes:
+    target = b[0]
+    if af == 6:
+      target = re.sub('%[^ ]+/','/',target)
+      print('blackhole ' + target)
+    else:
+      cidr = cidr_from_netstat_target(target)
+      print('blackhole ' + cidr)
   return True
 
+def cidr_from_netstat_target(target):
+  dots=target.count('.')
+  if target.find('/') == -1:
+    addr=target
+    netmask=8+dots*8
+  else:
+    [addr,netmask] = target.split('/')
+
+  if dots == 2:
+    addr = addr + '.0'
+  elif dots == 1:
+    addr = addr + '.0.0'
+  elif dots == 0:
+    addr = addr + '.0.0.0'
+  return addr + '/' + str(netmask)
+
+
 def do_route_add(argv,af):
   target=argv[0]
   via=argv[1]
-  gw=argv[2]
+  options=""
```

Comment:
There a number of options/flags that could be passed in to `route` so I set it up for more use